ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -3742,6 +3755,80 @@ private Operator genNotNullFilterForJoinSourcePlan(QB qb, Operator input,
     return output;
   }
 
+  /** This function is used for the purpose of adding an
+   *  outerQueryTable.outerQueryCol is not null predicate to the join,
+   *  since it is not added naturally like it is done in the case of "IN" operator
+   **/
+  private Operator genNotNullFilterForJoinSourcePlanNIC(QB qb, Operator input,
```

Comment:
Too many duplicate code. Can simplify this.

Reply:
done